crate/agent/src/utils.rs

```diff
@@ -56,3 +75,80 @@ pub fn _filter_whilelist(entry: &DirEntry) -> Result<bool, Error> {
 
     Ok(true)
 }
+
+/// Generate a self-signed certificate
+pub fn generate_self_signed_cert(
+    subject: &str,
+    subject_alternative_names: Vec<&str>,
```

Comment:
no need to take ownership
```suggestion
    subject_alternative_names: &[&str],
```